core/src/messages/agent.rs

```diff
@@ -254,6 +255,49 @@ pub struct SpawnRequest {
     #[serde(default = "ResourceLimits::default")]
     pub resource_limits: ResourceLimits,
 }
+*/
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct DockerExecutableConfig {
+    /// The container image to run.
+    pub image: String,
+
+    /// The timeout after which the drone is shut down if no connections are made.
+    #[serde_as(as = "DurationSeconds")]
+    pub max_idle_secs: Duration,
+
+    /// Environment variables to pass in to the container.
+    pub env: HashMap<String, String>,
+
+    /// Credentials used to fetch the image.
+    pub credentials: Option<DockerCredentials>,
+
+    /// Resource limits
+    #[serde(default = "ResourceLimits::default")]
+    pub resource_limits: ResourceLimits,
+}
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct RouterConfig {
```

Comment:
I'm inclined to call this `ProxyConfig`, and use it for things that are specific to the proxy. Currently, that's just the `backend_id`, but will include auth tokens (#118) eventually as well.

We could also just leave these fields in the `SpawnRequest` itself -- unlike the executor, I don't see a clear advantage to bundling them together at this point.

Reply:
I think that's sensible yes, will do (leaving the other fields in the SpawnRequest itself). The current main motivation after all is to have `DockerExecutableConfig` be seperately serializeable.